pkg/tls/tls.go

```
@@ -0,0 +1,185 @@
+// via unstructured client and returns TLS option functions for controller-runtime.
+// On non-OpenShift clusters or when the profile cannot be read, it returns
+// hardened Intermediate defaults. Returns an error only on unexpected failures
+// that should prevent startup (fail-closed).
+func Resolve(ctx context.Context, cfg *rest.Config) (Result, error) {
+	var result Result
+
+	k8sClient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		log.Info("Failed to create bootstrap client for TLS profile, using hardened defaults")
+		result.TLSOpts = append(result.TLSOpts, intermediateWithALPN)
+		return result, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 80) but it returns nil

(nilerr)